stablehlo/tools/CMakeLists.txt

```diff
@@ -38,14 +38,14 @@ mlir_check_all_link_libraries(stablehlo-opt)
 add_llvm_executable(stablehlo-interpreter StablehloInterpreterMain.cpp)
 llvm_update_compile_flags(stablehlo-interpreter)
 target_link_libraries(stablehlo-interpreter PRIVATE
-  MLIRDataLayoutInterfaces
   MLIRFuncDialect
   MLIRIR
   MLIRSupport
   MLIRTranslateLib
   StablehloOps
   StablehloReferenceInterpreter
   StablehloReferenceTensor
+  MLIRDataLayoutInterfaces    # StablehloOps references MLIRDataLayoutInterfaces
```

Comment:
Order dependent includes/dependencies usually indicates a problem elsewhere. Not sure if that rule applies to cmake files.

It is unclear why `stablehlo/tools/CMakeLists.txt` needs `MLIRDataLayoutInterfaces`, it looks like the only use of DataLayout is in `StablehloOps` as noted.

Could we move this to the dependencies of `StablehloOps` in `stablehlo/dialect/CMakeLists.txt`?

```
add_mlir_dialect_library(StablehloOps
  PARTIAL_SOURCES_INTENDED
  StablehloOps.cpp
  StablehloBytecode.cpp

  DEPENDS
  StablehloOpsIncGen

  LINK_LIBS PUBLIC
  StablehloBase
  MLIRArithmeticDialect
  MLIRDataLayoutInterfaces
  MLIRIR
  ...
)
```

I tried building with this change and everything worked locally.

Reply:
Nice! Glad that worked, marking as resolved. I'll leave the final approval to @burmako. 

Appreciate the fix!